src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using Newtonsoft.Json;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.ToolManifest
+{
+    internal class ToolManifestFinder : IToolManifestFinder
```

Comment:
@dsplaisted @nguerrera all logic about the file format is in this single. If we want to switch to lisp tomorrow, only this need to change